CMakeLists.txt

```diff
@@ -1,8 +1,37 @@
 cmake_minimum_required(VERSION 3.12)
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake ${CMAKE_MODULE_PATH})
+
+
+###
+###  Semantic versioning
+###
+include ( GetGitRevisionDescription )
+#git_describe(VERSION)
+git_get_exact_tag(VERSION)
+if(NOT VERSION)
+  message("No Git tag version found setting v0.0.0")
+  set(VERSION "v0.0.0")
+endif(NOT VERSION)
+
+
+message( ${VERSION})
+
+
+get_git_head_revision(REFSPEC HASHVAR)
+
+string(REGEX REPLACE "^v([0-9]+)\\..*" "\\1" MLSPP_VERSION_MAJOR "${VERSION}")
+string(REGEX REPLACE "^v[0-9]+\\.([0-9]+).*" "\\1" MLSPP_VERSION_MINOR "${VERSION}")
+string(REGEX REPLACE "^v[0-9]+\\.[0-9]+\\.([0-9]+).*" "\\1" MLSPP_VERSION_PATCH "${VERSION}")
+string(REGEX REPLACE "^v[0-9]+\\.[0-9]+\\.[0-9]+(.*)" "\\1" MLSPP_VERSION_SHA1 "${VERSION}")
+mark_as_advanced(MLSPP_VERSION_MAJOR MLSPP_VERSION_MINOR MLSPP_VERSION_PATCH)
```

Comment:
In the interest of making these CMakeLists reusable, I would avoid using `MLSPP` in variable names. Suggest `LIB_VERSION` etc.